Added iOS support

MHDtA-dev opened the pull request:
I was able to run wgpu on iOS using your C++ header and the SDL2 library. To do this, I added support for creating a Metal layer for UIKit to this library.

![imgA](https://github.com/eliemichel/sdl2webgpu/assets/56554365/53611de7-475c-41b0-b458-d78c4f03766e)
![imgB](https://github.com/eliemichel/sdl2webgpu/assets/56554365/c13a90ff-c01b-45a3-8a89-8c4ef64d655c)

eliemichel commented:
Thanks! Glad to see it working on iOS :) Any reason why you needed to add a dedicated header file although the function defined there has the same signature?

MHDtA-dev commented:
This header was created automatically by Xcode, when I created sdl2webgpu_ios.mm file. I didn't include sdl2webgpu.h and sdl2webgpu.c in my project, so I implemented this function again and declared it in ios header. Then I just added ios files to this project. I think it is really possible to use existing header.

eliemichel commented:
Okey I see! I tried to merge into the same file, could you try the `eliemichel/ios` branch? https://github.com/eliemichel/sdl2webgpu/tree/eliemichel/ios

MHDtA-dev commented:
It seems this code doesn't want to work in .c file, I think we have to move it to .mm

![screensh](https://github.com/eliemichel/sdl2webgpu/assets/56554365/fdaef854-3a01-4462-be35-7b71cff6d2f2)

MHDtA-dev commented:
Nah, I was able to run it by changing file type to "Objective-C++ source" in "Identity and Type" tab. But for some reason it gave me errors:


sdl2webgpu.c:90:13 Taking the address of a temporary object of type 'WGPUSurfaceDescriptor'
sdl2webgpu.c:92:58 Taking the address of a temporary object of type 'WGPUSurfaceDescriptorFromMetalLayer'

I replaced 

```
return wgpuInstanceCreateSurface(
            instance,
            &(WGPUSurfaceDescriptor){
                .label = NULL,
                .nextInChain = (const WGPUChainedStruct*)&(WGPUSurfaceDescriptorFromMetalLayer) {
                    .chain = (WGPUChainedStruct){
                        .next = NULL,
                        .sType = WGPUSType_SurfaceDescriptorFromMetalLayer,
                    },
                    .layer = (__bridge void*)metal_layer,
                },
            }
        );
```
with

```

WGPUSurfaceDescriptorFromMetalLayer metalLayerDesc = {
            .chain = {
                .next = NULL,
                .sType = WGPUSType_SurfaceDescriptorFromMetalLayer,
            },
            .layer = (__bridge void*)metal_layer,
        };

        WGPUSurfaceDescriptor surfaceDesc = {
            .label = NULL,
            .nextInChain = (const WGPUChainedStruct*)&metalLayerDesc,
        };

        return wgpuInstanceCreateSurface(instance, &surfaceDesc);

```

And it worked

eliemichel commented:
> Nah, I was able to run it by changing file type to "Objective-C++ source" in "Identity and Type" tab.

This was supposedly handled by the CMakeLists, did you use CMake or configure the project manually in your IDE?

> I replaced

Thanks for the hint, I updated the branch!

MHDtA-dev commented:
I've never used CMake in iOS development, and I even don't know if it is possible to do. iOS is not my main target platform and I prefer classical development in Xcode without any additional tools. For desktop of course I use CMake and my apps build perfectly with such libraries.

eliemichel commented:
Ok this is interesting insight (I never did iOS development, and never owned an iPhone), could you summarize exactly how to set up this file in the wording that feels most accurate to you, so that we can include it in teh readme? :)

MHDtA-dev commented:
Very soon I'm going to write an article for learn webgpu on how to set up webgpu for iOS, it will also include instructions for setting up sdl2webgpu.

eliemichel commented:
Squashed and merged through 7f8cd3c19ec0d583f7331576b54b8aa8cbeefa00
